app/queries/filtered_notes_query.rb

```diff
@@ -0,0 +1,27 @@
+class FilteredNotesQuery
+  ALLOWED_PARAMS = %i[query].freeze
+
+  attr_reader :relation, :filter_params
+  private :relation, :filter_params
+
+  def initialize(relation, filter_params = {})
+    @relation = relation
+    @filter_params = filter_params
+  end
+
+  def all
+    filter_params.slice(*ALLOWED_PARAMS).reduce(relation) do |relation, (key, value)|
+      next relation if value.blank?
+
+      send("by_#{key}", relation, value)
+    end
+  end
+
+  private
+
```

Comment:
I know that it was my code example, but you can move `attr_reader` here from line #4 to avoid `private` word duplication.

```ruby
private

attr_reader :relation, :filter_params
```